numpy/lib/tests/test_arraypad.py

```diff
@@ -1121,6 +1121,21 @@ def test_check_02(self):
         a = np.pad([1, 2, 3], 4, 'wrap')
         b = np.array([3, 1, 2, 3, 1, 2, 3, 1, 2, 3, 1])
         assert_array_equal(a, b)
+    
+    def test_check_03(self):
+        a = np.arange(9, dtype=float).reshape(3, 3)
+        a = np.pad(a, [(2, 4), (1, 1)], mode='wrap')
+        b = np.array([
+            [5, 3, 4, 5, 3],
+            [8, 6, 7, 8, 6],
+            [2, 0, 1, 2, 0],
+            [5, 3, 4, 5, 3],
+            [8, 6, 7, 8, 6],
+            [2, 0, 1, 2, 0],
+            [5, 3, 4, 5, 3],
+            [8, 6, 7, 8, 6],
+            [2, 0, 1, 2, 0]])
+        assert_array_equal(a, b)
```

Comment:
If we use a non-minimal two dimensional test case, we can test for the fault behavior in both dimension. Also might be good to add a comment explaining what this tests does.

```suggestion
        # Assert that 'wrap' pads only with multiples of the original area if
        # the pad width is larger than the original array
        a = np.arange(4).reshape(2, 2)
        a = np.pad(a, [(1, 3), (3, 1)], mode='wrap')
        b = np.array(
            [[3, 2, 3, 2, 3, 2],
             [1, 0, 1, 0, 1, 0],
             [3, 2, 3, 2, 3, 2],
             [1, 0, 1, 0, 1, 0],
             [3, 2, 3, 2, 3, 2],
             [1, 0, 1, 0, 1, 0]]
        )
        assert_array_equal(a, b)
```

Reply:
One thing you could do, is use `np.tile(arr, (3, 3))` or even `(5, 5)`.  Then do some slicing.
Then you could use `@pytst.mark.parametrize` to test a could of different pad widths.

Reply:
I implemented a version using `np.tile` and `@pytest.mark.parametrize`, but it seems the test case becomes less straightforward, since it needs some computation for slice index. So, I would prefer still using Lagru's case.